### ISSUE TITLE
Tokio spawn memory leak

### ISSUE THREAD

**Issue by alexandermake** [tokio-rs/tokio#6065]
Rust version:
rustc 1.73.0 (cc66ad468 2023-10-03)
 
```
│   ├── tokio v1.33.0
│   │   │   └── tokio-macros v2.1.0 (proc-macro)
│   │   ├── tokio-util v0.7.9
│   │   │   ├── tokio v1.33.0 (*)
│   │   ├── tokio v1.33.0 (*)
│   │   ├── tokio v1.33.0 (*)
│   │   └── tokio-rustls v0.24.1
│   │       └── tokio v1.33.0 (*)
│   │   ├── tokio v1.33.0 (*)
│   │   └── tokio-native-tls v0.3.1
│   │       └── tokio v1.33.0 (*)
│   ├── tokio v1.33.0 (*)
│   ├── tokio-native-tls v0.3.1 (*)
│   ├── tokio-rustls v0.24.1 (*)
└── tokio v1.33.0 (*)
```

OS: Ubuntu 22.04 (Linux 5.15.57-051557-generic #202207231132 SMP Sat Jul 23 11:50:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux)

[Repository](https://github.com/alexandermake/spawn)

Memory is never freed, every new task add around 10mb and this cause memory leak in large projects.
Reproduction with repo: cargo run

https://github.com/tokio-rs/tokio/discussions/4678
https://github.com/tokio-rs/tokio/issues/4406

**Comment by Darksonn** [comment]
Most likely, your tasks are not actually exiting because the requests are being blocked by tokio.rs DOS detection, causing them to hang until your timeout. Tasks that have not exited will indeed consume memory. That's normal.

**Comment by Originalimoc** [comment]
What's the point of using yield_now()?
